### ISSUE TITLE
Coverage inaccuracy

### ISSUE THREAD

**Issue by bramtayl** [JuliaLang/julia#35395]
Apologies if there's already a corresponding open issue for this. I'm running into coverage issues. The coverage reports for 

https://github.com/bramtayl/JuliennedArrays.jl

and especially https://github.com/bramtayl/LightQuery.jl

list several lines of code as uncovered, and several lines of code are considered to be not code. As far as I can tell, the true coverage statistic for both packages should be 100%, that is, if I comment out any line of code and replace it with `error()`, tests will fail.

**Comment by ViralBShah** [comment]
Close?

**Comment by bramtayl** [comment]
I don't think this solved. Ref https://github.com/JuliaLang/julia/pull/44455#issuecomment-1060079781

**Comment by brenhinkeller** [comment]
Is this an issue for Julia or an issue for codecov.io?

**Comment by vtjnash** [comment]
For julia. Usually it is constant-folded values that get evaluated, but not run so they end up not getting counted

**Comment by dhanak** [comment]
I ran into the same issue with my library [InlineDispatch.jl](https://github.com/dhanak/InlineDispatch.jl). There are 7 lines ([4 here](https://app.codecov.io/gh/dhanak/InlineDispatch.jl/blob/master/src%2FInlineDispatch.jl#L16) and [3 here](https://app.codecov.io/gh/dhanak/InlineDispatch.jl/blob/master/src%2FInlineDispatch.jl#L36)) reported as uncovered code lines, whereas they are part of a docstring. There's also a line in the body of the macro which is shown as uncovered, but it is most definitely covered, otherwise the tests wouldn't work ([here](https://app.codecov.io/gh/dhanak/InlineDispatch.jl/blob/master/src%2FInlineDispatch.jl#L73))

This results in the coverage being reported as 58%, whereas it should be 100%.

Note that this is not a codecov issue, the same anomaly can be observed in the cov files generated by `Pkg.test(coverage = true)`.

**Comment by MilesCranmer** [comment]
This popped up on my Google search so I just wanted to document my workaround. There is a (fairly new) [directive in Coveralls.jl](https://github.com/JuliaCI/Coverage.jl?tab=readme-ov-file#exclude-specific-lines-or-sections-from-coverage) to disable coverage for some code:

> To exclude specific code blocks, surround the section with `COV_EXCL_START` and `COV_EXCL_STOP` comments:
> 
> ```julia
> # COV_EXCL_START
> foo() = nothing
> # COV_EXCL_STOP
> ```
> 
> To exclude a single line, add a comment with `COV_EXCL_LINE`:
> ```julia
> const a = 1 # COV_EXCL_LINE
> ```

**Comment by maleadt** [comment]
> a (fairly new) [directive in Coverage.jl](https://github.com/JuliaCI/Coverage.jl?tab=readme-ov-file#exclude-specific-lines-or-sections-from-coverage)

This was added in https://github.com/JuliaCI/CoverageTools.jl/pull/35, which is almost 5 years old, so I wouldn't call it fairly new.

**Comment by MilesCranmer** [comment]
Oops! Newly discovered by me
